ui/app/src/components/FormsEngine/controls/Slug.tsx

```
@@ -15,59 +15,183 @@
+	const formContext = useStableFormContext();
+	const isNewForm = nnou(formContext.props?.create);
 	const htmlId = useId();
 	if (field.id === 'fileName') {
 		throw new Error('Detected field ID "fileName" instead "file-name" at the "Slug" Control.');
 	}
-	// It'd be better to avoid this. A specific control should refer to a specific field ID/property all the time.
-	const fieldId = field.id === 'file-name' ? 'folder-name' : field.id;
+	const isPage = contentType?.type === 'page';
+	const [isContentAsFolder, setIsContentAsFolder] = useState<boolean>(isPage);
+	const isFolder = field.id === 'file-name' && isPage && isContentAsFolder;
+	const isFolder = field.id === 'file-name' && isPage && contentAsFolder;
```

Biome (2.1.2)

[error] Shouldn't redeclare 'isFolder'. Consider to delete it or rename it.

'isFolder' is defined here:


(lint/suspicious/noRedeclare)